bazel/android_artifacts.bzl

```diff
@@ -200,7 +200,24 @@ def _create_jni_library(name, native_deps = []):
         srcs = native_deps,
     )
 
-    return jni_archive_name + "_unsigned.apk"
+    # Depend on the explicit envoy_aar_jni_unsigned.apk which contains the .so files.
```

Comment:
The main reason for this addition is to allow us to rename any or all of the `cc_binary` targets which output the `.so` file. The current setup forces us to fixate the name but this will allow us to change the end resulting so file names to be different than the `cc_binary` target name.